autonomy/cli/scaffold_fsm.py

```diff
@@ -167,635 +155,193 @@ def payloads(self) -> List[str]:
         """Payloads"""
         return [s.replace(ROUND, PAYLOAD) for s in self.rounds]
 
+    def _parse_transition_func(self) -> Dict[str, Dict[str, str]]:
+        """Parse the transition function from the spec to a nested dictionary."""
 
-class RoundFileGenerator(AbstractFileGenerator):
-    """File generator for 'rounds.py' modules."""
+        result: Dict[str, Dict[str, str]] = {}
+        for (round_cls_name, event_name), value in self.dfa.transition_func.items():
+            result.setdefault(round_cls_name, {})[f"{EVENT}.{event_name}"] = value
+        for state in self.dfa.states:
+            if state not in result:
+                result[state] = {}
+        return result
 
-    FILENAME = ROUNDS_FILENAME
+    @property  # TODO: functools cached property
+    def template_kwargs(self) -> Dict[str, str]:
+        """All keywords for string formatting of templates"""
```

Comment:
one dictionary with all kwargs that are non-specific (not round, behaviour or payload specific)